.editorconfig

```diff
@@ -4,31 +4,5 @@ root = true
 charset = utf-8
 end_of_line = lf
 insert_final_newline = true
-tab_width = 4
 trim_trailing_whitespace = false
```

Comment:
trim trailing whitespace should be enabled

Reply:
trim trailing whitespace often results in messy commits as I intentionally leave trailing whitespace up to the indent of the block that contains it
e.g.
```java
public void foo() {
  //vertical whitespace here is indented at the block indent level
}
```

Reply:
👍